### PR TITLE
Remove NewGameChooserMapCache.

### PR DESCRIPTION
**Pull request by DanVanAtta**
Current code reloads the maps anyways, so it does not provide benefit. That is an artifact of the change we made to reload the maps directly after download, so relatively recently we stopped using the cache.

---
This change is [Reviewable](https://reviewable.io/reviews/triplea-game/triplea/693)

### CODE REVIEW

File: src/games/strategy/engine/framework/startup/launcher/AbstractLauncher.java [hunk 1]
```diff
@@ -39,14 +39,7 @@ public void launch(final Component parent) {
     if (!m_headless && !SwingUtilities.isEventDispatchThread()) {
       throw new IllegalStateException("Wrong thread");
     }
-    final Runnable r = new Runnable() {
-      @Override
-      public void run() {
-        // we don't want to keep around all the memory for this, since we have the gamedata that we want
-        NewGameChooser.clearNewGameChooserModel();
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/startup/launcher/AbstractLauncher.java]: removed this one line, then collapsed into a one-line lambda

File: src/games/strategy/engine/framework/ui/NewGameChooser.java [hunk 2]
```diff
@@ -29,19 +29,14 @@
...
-  private static ClearGameChooserCacheMessenger cacheClearedMessenger;
 
   private JButton okButton;
   private JButton cancelButton;
   private JList<NewGameChooserEntry> gameList;
   private JPanel infoPanel;
   private JEditorPane notesPanel;
   private NewGameChooserModel gameListModel;
-  private NewGameChooserEntry choosen;
+  private NewGameChooserEntry chosen;
```
Comment (DanVanAtta) [inline src/games/strategy/engine/framework/ui/NewGameChooser.java]: opportunistic fix of this variable name, while we were modifying this class and lines that related to 'choosen'

**Comment by RoiEXLab** [comment]
This will conflict with my ongoing PR.. this is good, since that means fewer mistakes are possible ^^

**Comment by RoiEXLab** [comment]
Before Merging:
Please remove `import com.google.common.collect.ImmutableSet;` in `NewGameChooserModel.java`
and `boolean refreshedAlready = false;` in `loadDefaultGame(*)` in `GameSelectorModel.java`,
since those are not used anymore...

**Comment by RoiEXLab** [comment]
And also... it's just a removal of uneccesary code - I just tested it...
Same Error as always...

**Comment by DanVanAtta** [comment]
No merge conflicts on this one. Ready to review

**Comment by ron-murhammer** [comment]
@DanVanAtta Looks good was just letting you address @RoiEXLab comments.

**Comment by DanVanAtta** [comment]
Resolved feedback in: 5f1fb87, relatively simple fixes, so going to go ahead with the merge

**Pull request merged**